index.bs

```diff
@@ -155,114 +155,120 @@ mean the same thing as:
 
-When Reflect.Loader is called with no arguments, the following steps are taken:
+<h4 id="new-loader">Loader()</h4>
+
+When Loader is called with no arguments, the following steps are taken:
 
 <emu-alg>
 1. If NewTarget is *undefined*, then throw a *TypeError* exception.
-1. Let _O_ be OrdinaryCreateFromConstructor(NewTarget, "Reflect.Loader.prototype").
+1. Let _O_ be OrdinaryCreateFromConstructor(NewTarget, "Loader.prototype").
 1. ReturnIfAbrupt(_O_).
-1. Let _registry_ to a new Registry(_O_).
-1. Set _O_’s [[Registry]] internal slot to _registry_.
+1. Set _O_’s [[Registry]] internal slot to CreateRegistry().
 1. Return _O_.
 </emu-alg>
 
-<h3 id="properties-of-the-loader-constructor">Properties of the Reflect.Loader Constructor</h3>
+<h3 id="properties-of-the-loader-constructor">Properties of the Loader Constructor</h3>
 
-The value of the \[[Prototype]] internal slot of the Reflect.Loader constructor is the intrinsic object %FunctionPrototype%.
+The value of the \[[Prototype]] internal slot of the Loader constructor is the intrinsic object %FunctionPrototype%.
 
-Besides the internal slots and the length property (whose value is 0), the Reflect.Loader constructor has the following properties:
+Besides the internal slots and the length property (whose value is 0), the Loader constructor has the following properties:
 
-<h4 id="Reflect.Loader.prototype">Reflect.Loader.prototype</h4>
+<h4 id="Loader.prototype">Loader.prototype</h4>
 
-The value of Reflect.Loader.prototype is an ordinary object.
+The value of Loader.prototype is an ordinary object.
 
 This property has the attributes { \[[Writable]]: false, \[[Enumerable]]: false, \[[Configurable]]: false }.
 
-<h3 id="sec-properties-of-the-loader-prototype-object">Properties of the Reflect.Loader Prototype Object</h3>
+<h3 id="sec-properties-of-the-loader-prototype-object">Properties of the Loader Prototype Object</h3>
 
-<h4 id="Reflect.Loader.prototype.constructor">Reflect.Loader.prototype.constructor</h4>
+<h4 id="Loader.prototype.constructor">Loader.prototype.constructor</h4>
 
-The initial value of Reflect.Loader.prototype.constructor is Reflect.Loader.
+The initial value of Loader.prototype.constructor is Loader.
 
-<h4 id="reflect-loader-import">Reflect.Loader.prototype.import(name[, referrer])</h4>
+<h4 id="loader-import">Loader.prototype.import(name[, referrer])</h4>
 
 The following steps are taken:
 
 <emu-alg>
 1. Let _loader_ be *this* value.
-1. If Type(_loader_) is not Object, throw a *TypeError* exception.
-1. If _loader_ does not have a [[Registry]] internal slot throw a *TypeError* exception.
+1. Assert: _loader_ must be an instance of a concrete subclass of Loader.
```

Comment:
This isn't what "Assert" means in the 262 spec. It's an internal assertion about the coherence of the spec itself (intuitively, a failed Assert means the JS engine should crash/panic/segfault). A check that results in a JS exception should be done with an explicit *TypeError* exception.

We should tell bterlson that this deserves an abstraction to avoid boilerplate, BTW. But it's separate from Assert.